imports/plugins/included/discount-codes/server/methods/methods.js

```diff
@@ -60,14 +89,42 @@ export const methods = {
    * @param  {String} id cart id of which to remove a code
    * @param  {String} codeId discount Id from cart.billing
    * @param  {String} collection collection (either Orders or Cart)
+   * @param  {String} [token] Cart or order token if anonymous
    * @return {String} returns update/insert result
    */
-  "discounts/codes/remove"(id, codeId, collection = "Cart") {
+  "discounts/codes/remove"(id, codeId, collection = "Cart", token) {
     check(id, String);
     check(codeId, String);
     check(collection, String);
+    check(token, Match.Maybe(String));
+
     const Collection = Reaction.Collections[collection];
 
+    if (collection === "Cart") {
+      let { cart } = getCart(id, { cartToken: token, throwIfNotFound: true });
+
+      // If we found a cart, then the current account owns it
+      if (!cart) {
+        cart = Collection.findOne({ _id: id });
+        if (!cart) {
+          throw new ReactionError("not-found", "Cart not found");
+        }
+
+        if (!Reaction.hasPermission("discounts", Reaction.getUserId(), cart.shopId)) {
+          throw new ReactionError("access-denied", "Access Denied");
+        }
+      }
+    } else {
+      const order = Collection.findOne({ _id: id });
+      if (!order) {
+        throw new ReactionError("not-found", "Order not found");
+      }
```

Comment:
Similar question to the one I asked previously about checking for a product before checking permissions. I know in that case, you said the product ID was public knowledge, so it was OK. In this case, is the order or cart ID public knowledge? Should we check permissions first? I see that the permission check is for `discounts` and not `orders` or `cart`, so maybe it's irrelevant here.